torch/_inductor/lowering.py

```diff
@@ -2780,6 +2780,9 @@ def gather(x, dim, index, sparse_grad=False):
     # sparse_grad doesn't affect forward computation,
     # and backward tracing is taken care of by AOT Autograd
     assert isinstance(x, TensorBox)
+    if index.get_numel() == 0:
+        return new_empty(x, [0])
```

Comment:
Shouldn't this be `index.get_shape()`? Also what fails when using the `Pointwise` node with 0 numel? Ideally that would work gracefully.